pywbem_mock/_baseprovider.py

```diff
@@ -41,34 +41,22 @@
 from __future__ import absolute_import, print_function
 
 from pywbem import CIMError, \
-    CIM_ERR_NOT_FOUND, \
+    CIM_ERR_NOT_FOUND, CIMInstance, CIMClass, \
     CIM_ERR_ALREADY_EXISTS, CIM_ERR_INVALID_NAMESPACE, \
-    CIM_ERR_NAMESPACE_NOT_EMPTY
+    CIM_ERR_NAMESPACE_NOT_EMPTY, \
+    WBEMServer
+
+from pywbem._nocasedict import NocaseDict
 
 from pywbem._utils import _format
 
 # pywbem_mock implementation configuration variables that are used in
 # request responsders.
 
-# Default Max_Object_Count for Fake Server if not specified by request
-_DEFAULT_MAX_OBJECT_COUNT = 100
-
-# Maximum Open... timeout if not set by request
-OPEN_MAX_TIMEOUT = 40
-
 # Per DSP0200, the default behavior for EnumerateInstance DeepInheritance
 # if not set by server.  Default is True.
 DEFAULT_DEEP_INHERITANCE = True
 
-# Value of LocalOnly parameter for instance retrevial operations.  This is
-# set to False in this implementation because of issues between different
-# versions of the DSP0200 specification that defined incompatible behavior
-# for this parameter that were resolved in Version 1.4 by
-# stating that False was the recommended Server setting for all instance
-# retrevial requests and that clients should use False also to avoid the
-# incompatibility.
-INSTANCE_RETRIEVE_LOCAL_ONLY = False
-
 
 class BaseProvider(object):
```

Comment:
Should this class be an abstract base class, if only to make it explicit that it has no objects? That would also allow removing its repr() method.

Reply:
TODO

Reply:
That is a messy question.  It is not really an abstract class in that it does not dictate any abstract methods but right now it is more than a mixin in that we push the cimrepository all the way up to here.

Late afternoon is no time for me to make a decision on this.  Marked TODOTHISFIX  and

DISCUSSION

Reply:
To me, the purpose of an abstract class is not only to define abstract methods, but also to indicate that it does not make sense to instantiate it. I was trying to find some evidence for that view, but what I found about abstract classes in Python and in general was quite silent about this case.

I think we should probably go for the more obvious solution to use ABCs only if we need to define abstract methods. I.e. not in this case.

Reply:
I agree which means that BaseProvider is kind of a strange form.  It is abstract in the sense that it is not used directly but real in the sense that it has an __init__ method.

DISCUSSION

Reply:
DONE - We agreed to withdraw this comment.